tests/unit/s2n_handshake_test.c

```diff
@@ -82,30 +85,30 @@ static int try_handshake(struct s2n_connection *server_conn, struct s2n_connecti
 }
 
 int test_cipher_preferences(struct s2n_config *server_config, struct s2n_config *client_config) {
-    const struct s2n_cipher_preferences *cipher_preferences;
-
-    cipher_preferences = server_config->cipher_preferences;
-    notnull_check(cipher_preferences);
+    struct s2n_cipher_preferences cipher_preferences = { 0 };
+    int default_cipher_suites_count = 0;
+    struct s2n_cipher_suite *default_cipher_suites[S2N_CIPHER_SUITE_COUNT];
 
     if (s2n_is_in_fips_mode()) {
         /* Override default client config ciphers when in FIPS mode to ensure all FIPS
          * default ciphers are tested.
          */
-        client_config->cipher_preferences = cipher_preferences;
-        notnull_check(client_config->cipher_preferences);
+        client_config->cipher_preferences = server_config->cipher_preferences;
```

Comment:
I'd love a close second look at this part. I don't THINK I changed the test's behavior, but I'm not entirely sure. Why do we set the client preferences to the server preferences when FIPS mode is enabled? How does that ensure we're testing all FIPS default ciphers?